portal/src/graphql/adminapi/AuditLogScreen.tsx

```diff
@@ -149,13 +175,44 @@ const AuditLogScreen: React.FC = function AuditLogScreen() {
         .toJSDate()
         .toISOString();
     }
-    return lastUpdatedAt;
+    return lastUpdatedAt.toISOString();
   }, [rangeTo, lastUpdatedAt]);
 
-  const isCustomDateRange = rangeFrom != null || rangeTo != null;
+  const location = useLocation();
+
+  const today = useConst(new Date());
+  const isCustomDateRange =
+    rangeFrom != null ||
+    (rangeTo != null &&
+      DateTime.fromJSDate(rangeTo).toISODate() !==
+        DateTime.fromJSDate(today).toISODate());
 
   const { renderToString } = useContext(Context);
 
+  useEffect(() => {
+    const page = offset / pageSize + 1;
+    const params: URLSearchParamsInit = {};
+    if (rangeFrom != null) {
+      params["from"] = DateTime.fromJSDate(rangeFrom).toISODate();
+    }
+    params["to"] = rangeTo
+      ? DateTime.fromJSDate(rangeTo).toISODate()
+      : DateTime.fromJSDate(lastUpdatedAt).toISODate();
+    params["page"] = page.toString();
+    params["order_by"] = sortDirection;
+    params["activity_type"] = selectedKey;
+    setSearchParams(params);
+  }, [
+    offset,
+    sortDirection,
+    isCustomDateRange,
+    rangeTo,
+    rangeFrom,
+    lastUpdatedAt,
+    setSearchParams,
+    selectedKey,
+  ]);
```

Comment:
When the UI is showing "Full Date Range" then there should be no "to" param appearing in the query.